src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx

```diff
@@ -0,0 +1,623 @@
+                                value: e[1],
+                                proj: projects[i]
+                            };
+                            tmp.push(obj);
+                        }
+                    }
+                });
+                subprojects[projects[i]] = tmp;
+
+
+            }
+            var o = 0;
+            for (i = 0; i < projects.length; i++) {
+                var opt = subprojects[projects[i]];
+                const tmp = [];
+                for (o = 0; o < opt.length; o++) {
+                    var myobj = {};
+                    myobj.project = projects[i];
+                    myobj.subproject = opt[o];
+                    var tab = data.filter((e) => {
+                        if (e[0] === projects[i] && e[1] === opt[o].value) {
+                            return e;
+                        }
+                    });
+                    myobj.option = tab.map((e) => {
+                        if (e[0] === projects[i] && e[1] === opt[o].value) {
+                            var rObj = {};
+                            rObj.label = e[2];
+                            rObj.value = e[4];
+                            rObj.proj = e[1];
+                            rObj.cat = e;
+                            return rObj;
+                        }
+                    });
+                    tmp.push(myobj);
+                }
+                catmaster[i] = {
+                    catalogs: tmp,
+                    project: projects[i],
+                    subproject: opt
+                };
+            }
+
+            this.setMaster({catmaster, cols});
+
+        }).catch(
+            (reason) => {
+                console.error(`Failed to get catalog: ${reason}`, reason);
+            }
+        );
+    }
+
+    /**
+     * Sets the object with the table model fetched, each element consists of
+     * an Object {catalogs: Array[7], project: "WISE", subproject: Array[7]}, where
+     * Each 'subproject' element is an option for the drop-down list such as {label: "AllWISE Database", value: "AllWISE Database", proj: "WISE"} for ecample
+     * Each 'catalogs' element is {project: "WISE", subproject: Object, option: Array[7]}, where
+     *  Each 'option' is element with catalog name value and other attributes such as
+     *      {label: "AllWISE Source Catalog", value: "wise_allwise_p3as_psd", proj: "AllWISE Database", cat: Array[12]} and
+     *      Each 'cat' array ith element is the value representing the ith column name below
+     *      ["projectshort", "subtitle", "description", "server", "catname", "cols", "nrows", "coneradius", "infourl", "ddlink", "catSearchProcessor", "ddSearchProcessor"]
+     * @see master table file from http://irsa.ipac.caltech.edu/cgi-bin/Gator/nph-scan?mode=ascii
+     * @param {Object} master table model
+     * @returns {Void} set the state to include master table
+     */
+    setMaster(master = {}) {
+        this.setState(...this.state, {master});
```

Comment:
...this.setState is also not necessary here.  setState will do a partial update.  no need to recreate the complete state each time.
there is a replaceState if that's what you want to do.